### PR TITLE
[infrastructure] Implemented IDialogService

### LINKED ISSUE #5195: [Codebase] Use IDialogService for dialogs
**Is your feature request related to a problem?**
We currently don't have any solution for showing `ContentDialog`s in ViewModels. Creating an instance of a `ContentDialog` doesn't comply with MVVM since the ViewModel should now nothing about the UI and it's components.

**Describe the solution**
To solve that we could use a `IDialogService` to show dialogs instead.

**Describe alternatives you've considered**
no.

**Scope**

| Capability | Priority |
| :---------- | :------- |
| Use IDialogService to replace `ContentDialog` creation in ViewModels | Must |

**Additional comment**
I already have the code for that in Clipboard Canvas so it could be reused here, if assigned. For #2341

### PR DESCRIPTION
**Pull request by d2dyno1**
- Added IDialogService for showing dialogs.

Existing API:
```cs
var viewModel = new DialogViewModel();
var dialog = new SomeDialog(viewModel);
dialog.ShowAsync();
```

New API surface:
```cs
var viewModel = new DialogViewModel();
var dialog = DialogService.GetDialog(viewModel)
dialog.ShowAsync();
```

Closes #5195
Closes #8458

### CODE REVIEW

**Comment by yaichenbaum** [comment]
> Additional note: I've also removed unused Bitlocked dialog and DynamicDialog still hasn't been converted. We'll have to split it into separate dialogs and for generic ones, new, less complex DynamicDialog would have to be created

@d2dyno1 what are we doing for BitLocker?

**Comment by d2dyno1** [comment]
> > Additional note: I've also removed unused Bitlocked dialog and DynamicDialog still hasn't been converted. We'll have to split it into separate dialogs and for generic ones, new, less complex DynamicDialog would have to be created
> 
> @d2dyno1 what are we doing for BitLocker?

Nothing. That's why the dialog was removed

**Comment by yaichenbaum** [comment]
Are we at least using the system dialog?

**Comment by d2dyno1** [comment]
> Are we at least using the system dialog?

I can't test

**Comment by gave92** [comment]
@yair100 yeah we've been using the system dialog since a while.

**Comment by d2dyno1** [comment]
@gave92 @yair100 Could you check? 😁

File: src/Files.Uwp/Dialogs/BitlockerDialog.xaml [hunk 1]
```diff
@@ -1,47 +0,0 @@
...
-                </Setter.Value>
-            </Setter>
-        </Style>
-    </ContentDialog.PrimaryButtonStyle>
-
-    <Grid MinWidth="300">
-        <StackPanel Orientation="Vertical" Spacing="4">
-            <PasswordBox
-                x:Name="PasswordInput"
-                x:Uid="BitlockerDialogInputText"
```
Comment (yaichenbaum) [inline src/Files.Uwp/Dialogs/BitlockerDialog.xaml:40]: Can you remove related strings?

**Review by yair100 — APPROVED** [review]

**Pull request merged**